_rules/link-accessible-name-c487ae.md

```diff
@@ -153,99 +162,92 @@ When `link` is off screen.
 
 <map name="planetmap">
-	<area shape="rect" coords="0,0,82,126" href="sun.htm" alt="Sun" />
+	<area shape="rect" coords="0,0,30,100" href="sun.htm" alt="Sun" />
 </map>
 ```
 
-#### Passed Example 11
-
-`a` element where [accessible name][] is not empty.
-
-```html
-<a href="http://www.w3.org/WAI">:-)</a>
-```
-
 ### Failed
 
 #### Failed Example 1
 
-Image link with empty [accessible name][].
+This `a` element has an empty [accessible name][].
 
 ```html
-<a href="http://www.w3.org/WAI"><img src="#"/></a>
+<a href="http://www.w3.org/WAI"><img src="/test-assets/shared/w3c-logo.png" role="none" /></a>
 ```
 
 #### Failed Example 2
 
-Image link where image is [marked as decorative](#marked-as-decorative).
+This `a` element with an image [marked as decorative](#marked-as-decorative) has an empty [accessible name][].
 
 ```html
-<a href="http://www.w3.org/WAI"><img src="#" alt=""/></a>
+<a href="http://www.w3.org/WAI"><img src="/test-assets/shared/w3c-logo.png" alt="" /></a>
 ```
 
 #### Failed Example 3
 
-Link with icon inserted via font-awesome.
+This `a` element with an icon inserted via font-awesome has an empty [accessible name][].
 
 ```html
+<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.2.0/css/font-awesome.min.css" rel="stylesheet" />
 <a href="http://www.w3.org/WAI"><i class="fa fa-download"></i></a>
 ```
 
 #### Failed Example 4
 
-Link with image that has empty title.
+This `a` element with an `img` with an empty `title` has an empty [accessible name][].
 
 ```html
-<a href="http://www.w3.org/WAI"><img src="#" title=""/></a>
+<a href="http://www.w3.org/WAI"><img src="/test-assets/shared/w3c-logo.png" title="" /></a>
 ```
 
 #### Failed Example 5
 
-Link with image that has empty `aria-labelledby`.
+This `a` element with an `img` with an `aria-labelledby` has an empty [accessible name][].
 
 ```html
-<a href="http://www.w3.org/WAI"><img src="#" aria-labelledy="id1"/></a>
+<a href="http://www.w3.org/WAI"><img src="/test-assets/shared/w3c-logo.png" aria-labelledby="id1" /></a>
 <div id="id1"></div>
 ```
 
 #### Failed Example 6
 
-`aria-labelledby` references a non-existing id.
+This `a` element with an `img` with an `aria-labelledby` referencing a non-existing id has an empty [accessible name][].
 
 ```html
-<a href="http://www.w3.org/WAI"><img src="#" aria-labelledby="id1"/></a>
+<a href="http://www.w3.org/WAI"><img src="/test-assets/shared/w3c-logo.png" aria-labelledby="id1" /></a>
 ```
 
 #### Failed Example 7
 
-[Non-visible](#visible) link.
+This `a` element placed off screen has an empty [accessible name][].
 
 ```html
 <a href="http://www.w3.org/WAI" style="left: -9999px; position: absolute;">
-	<img src="#" />
+	<img src="/test-assets/shared/w3c-logo.png" />
 </a>
 ```
 
 #### Failed Example 8
 
-Link is completely empty, but still shows up in focus order, so it should have a non-empty [accessible name][].
+This `a` element, which shows up in the focus order, has an empty [accessible name][]
```

Comment:
The rule has nothing to do with focus order, making this description a bit weird.
I understand it as "this link has no accessible name. Sure, it is totally empty and so you may think it doesn't need one, but it's still in focus order (or link list from AT), so it needs an accessible name or it will be very confusing for AT users".

I think we should reformulate the description.